Login Test

anatestpro opened the pull request:
IDE gives me an error. I did the task correct. Why does it say **"no such element: Unable to locate element: {"method":"css selector","selector":"img[class='avatar']"}"**?? There is such an element. I tried to use xpath or other elements. Assert function is still not working. How can I fix it?

anatestpro committed:
Login Test

rajniss21 commented:
If the assertion you're checking (Assert.assertTrue(avatarIcon.isDisplayed())) is meant to verify the presence of an element on the next page after a successful login, it won't work as intended in its current form.

In Selenium, the isDisplayed() method checks if the element is currently displayed on the page that is currently loaded in the WebDriver instance. If the element you're trying to assert (avatarIcon) is on the next page that appears after logging in, you should first wait for that next page to load and then locate and assert the presence of the element on the new page.

To do this, you should:

Wait for the next page to load after clicking the login button.
Locate the avatarIcon element on the new page.
Perform the assertion on the avatarIcon element.

rajniss21 commented:
your logintest page should look like this:

```
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.chrome.ChromeOptions;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.Assert;
import org.testng.annotations.Test;


import java.time.Duration;

public class LoginTests extends BaseTest {

    @Test
    public void LoginValidEmailPasswordTest() throws InterruptedException {

//      Added ChromeOptions argument below to fix websocket error
        ChromeOptions options = new ChromeOptions();
        options.addArguments("--remote-allow-origins=*");

        WebDriver driver = new ChromeDriver(options);
        driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(10));

       String url = "https://qa.koel.app/";
       driver.get(url);

       //email and password fields

        WebElement emailField = driver.findElement(By.cssSelector("[type='email']"));
        WebElement passwordField = driver.findElement(By.cssSelector("[type='password']"));
        WebElement submitButton = driver.findElement(By.cssSelector("[type='submit']"));



        emailField.click();
        emailField.clear();
        emailField.sendKeys("anastasia.dovhal@testpro.io");

        passwordField.click();
        passwordField.clear();
        passwordField.sendKeys("te$t$tudent");

        submitButton.click();
        // Wait for the next page to load (you can use explicit wait here)
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
        WebElement avatarIcon = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("img[class='avatar']")));

        Assert.assertTrue(avatarIcon.isDisplayed());

        driver.quit();
    }
}

```

rajniss21 commented:
Hello,
I have checked your homework 16, but you have got this issue, please review your code and fix it,

Hope the above suggestion helps.

Best wishes from the HW team (Rajnish, rajnish@testpro.io)!

anatestpro committed:
HW16

anatestpro commented:
Hi! Please review my HW16.java file. I'm not sure why it gives me an error. Is something with the Gradle? Thank you!

anatestpro commented:
![Screenshot 2023-11-06 205813](https://github.com/Test-Pro-Bootcamp/A47-QA-Automation-Framework/assets/139082811/3332e999-1412-466d-a992-a07d1533e640)

rajniss21 commented:
Please provide the detailed error log here.

In my case test has been passed.
![Screenshot_328](https://github.com/Test-Pro-Bootcamp/A47-QA-Automation-Framework/assets/25093177/031ed4ad-0e4c-4abc-a2cb-9054cd35ecf9)

anatestpro commented:
All the information about the error can be seen on the screenshot. I tried to google it and it says something might be with gradle. How can i check if i have the updated version?

anatestpro commented:
[
![Screenshot 2023-11-07 131502](https://github.com/Test-Pro-Bootcamp/A47-QA-Automation-Framework/assets/139082811/c5c6d845-bd77-4272-b88d-9971a43e9bcc)
](url)

anatestpro commented:
![Uploading Screenshot 2023-11-07 131704.png…]()

anatestpro commented:
![Screenshot 2023-11-07 131704](https://github.com/Test-Pro-Bootcamp/A47-QA-Automation-Framework/assets/139082811/1f940f43-3805-476e-83ec-00e9c648a887)

anatestpro commented:
Hi! I really need your help. My tests stopped running since I updated my chrome browser. I've tried so many different methods (like changing webdriver version and selenium and testNG version in build.gradle). When I changed versions, it leads to maven files. I see so many recommendations regarding pom.xml and JUnit, but can't find much about gradle and testNG. I've spent three days trying to figure out. And I'm hopeless at this point. Please help me to figure it out.

It gives me this error: org.openqa.selenium.SessionNotCreatedException: Could not start a new session. Response code 500. Message: session not created: This version of ChromeDriver only supports Chrome version 114
Current browser version is 122.0.6261.94 with binary path /Applications/Google Chrome.app/Contents/MacOS/Google Chrome
Host info: host: 'Anastasias-MBP.socal.rr.com', ip: '2603:8000:6640:c6:6800:3e6:8d4d:dfd9%en0'
Build info: version: '4.18.1', revision: 'b1d3319b48'
System info: os.name: 'Mac OS X', os.arch: 'aarch64', os.version: '14.2.1', java.version: '17.0.9'
Driver info: org.openqa.selenium.chrome.ChromeDriver
________________________________
From: Rajnish Pradhan ***@***.***>
Sent: Tuesday, November 7, 2023 8:10 AM
To: Test-Pro-Bootcamp/A47-QA-Automation-Framework ***@***.***>
Cc: Anastasia Dovhal ***@***.***>; Author ***@***.***>
Subject: Re: [Test-Pro-Bootcamp/A47-QA-Automation-Framework] Login Test (PR #291)


Please provide the detailed error log here.

In my case test has been passed.
[Screenshot_328]<https://user-images.githubusercontent.com/25093177/281106037-031ed4ad-0e4c-4abc-a2cb-9054cd35ecf9.png>

—
Reply to this email directly, view it on GitHub<https://github.com/Test-Pro-Bootcamp/A47-QA-Automation-Framework/pull/291#issuecomment-1799049585>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/BBFDYO6TM4DKRZZMAVZ4PK3YDJMOJAVCNFSM6AAAAAA45UAYE6VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTOOJZGA2DSNJYGU>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

anatestpro committed:
added Assert statement

added Assert statement
